packages/utils/aria.ts

```diff
@@ -11,6 +12,37 @@ export const eventKeys = {
   delete: 46,
 }
 
+// TODO: refactor all event.keyCode to event.code
+export const EVENT_CODE = {
+  tab: 'Tab',
+  enter: 'Enter',
+  space: 'Space',
+  left: 'ArrowLeft',
+  right: 'ArrowRight',
+  top: 'ArrowTop',
+  down: 'ArrowDown',
+  esc: 'Esc',
+  delete: 'Delete',
+  backspace: 'Backspace',
+}
+
+const FOCUSABLE_ELEMENT_SELECTORS =`a[href],button:not([disabled]),button:not([hidden]),:not([tabindex="-1"]),input:not([disabled]),input:not([type="hidden"]),select,textarea`
+
+/**
+ * Determine if the testing element is visible on screen no matter if its on the viewport or not
+ */
+export const isVisible = (element: HTMLElement) => {
+  if (process.env.NODE_ENV === 'test') return true
+  const computed = getComputedStyle(element)
+  // element.offsetParent won't work on fix positioned
+  return computed.position === 'fix' ? false : element.offsetParent !== null
```

Comment:
When the child elements contains positioning attributes like `position: fixed`, the element will go out of the current document flow, which means it would be so hard to track on the `fixed` element (which might be invisible on the document and people with screenreader device might have access them), by processing it this way we should warning the user not to apply position `fixed` element within the trapped content. Let's keep it this way right now, we could expect modification against this one in upcoming iterations. I will mark this as a potential issue.